app/org/sagebionetworks/bridge/services/email/ConsentEmailProvider.java

```diff
@@ -0,0 +1,156 @@
+        MimeTypeEmailBuilder builder = new MimeTypeEmailBuilder();
+        
+        String subject = String.format(CONSENT_EMAIL_SUBJECT, study.getName());
+        builder.withSubject(subject);
+        
+        final String sendFromEmail = isNotBlank(study.getSupportEmail()) ?
+                String.format("%s <%s>", study.getName(), study.getSupportEmail()) : defaultSender;
+        builder.withSender(sendFromEmail);
+        
+        builder.withRecipient(user.getEmail());
+        Set<String> emailAddresses = commaListToSet(study.getConsentNotificationEmail());
+        for (String email : emailAddresses) {
+            builder.withRecipient(email);
+        }
+                
+        final String consentDoc = createSignedDocument();
+        
+        // Consent agreement as message body in HTML
+        final MimeBodyPart bodyPart = new MimeBodyPart();
+        bodyPart.setText(consentDoc, StandardCharsets.UTF_8.name(), SUB_TYPE_HTML);
+        builder.withMessageParts(bodyPart);
+        
+        // Consent agreement as a PDF attachment
+        // Embed the signature image
+        String consentDocWithSig = consentDoc.replace("cid:consentSignature",
+                "data:" + consentSignature.getImageMimeType() +
+                ";base64," + consentSignature.getImageData());
+        final byte[] pdfBytes = createPdf(consentDocWithSig);
+        final MimeBodyPart pdfPart = new MimeBodyPart();
+        DataSource source = new ByteArrayDataSource(pdfBytes, MIME_TYPE_PDF);
+        pdfPart.setDataHandler(new DataHandler(source));
+        pdfPart.setFileName("consent.pdf");
+        builder.withMessageParts(pdfPart);
+        
+        // Write signature image as an attachment, if it exists. Because of the validation in ConsentSignature, if
+        // imageData is present, so will imageMimeType.
+        // We need to send the signature image as an embedded image in an attachment because some email providers
+        // (notably Gmail) block inline Base64 images.
+        MimeBodyPart sigPart = null;
+        if (consentSignature.getImageData() != null) {
+            // Use pre-encoded MIME part since our image data is already base64 encoded.
+            sigPart = new PreencodedMimeBodyPart(HEADER_CONTENT_TRANSFER_ENCODING_VALUE);
+            sigPart.setContentID(HEADER_CONTENT_ID_CONSENT_VALUE);
+            sigPart.setHeader(HEADER_CONTENT_DISPOSITION, HEADER_CONTENT_DISPOSITION_CONSENT_VALUE);
+            sigPart.setContent(consentSignature.getImageData(), consentSignature.getImageMimeType());
+        }
+        builder.withMessageParts(sigPart);
+
+        return builder.build();
+    }
+
+    private String createSignedDocument() {
+        final String filePath = studyConsent.getPath();
+        final FileSystemResource resource = new FileSystemResource(filePath);
+        try (InputStreamReader isr = new InputStreamReader(resource.getInputStream(), StandardCharsets.UTF_8);) {
```

Comment:
This is not your change.  But I think it is safer to declare the input stream of resource.getInputStream() separately so that it is guaranteed its close() method will be called at the end of the try block.